axelor-human-resource/src/main/java/com/axelor/apps/hr/web/timesheet/TimesheetController.java

```diff
@@ -573,15 +574,44 @@ public void timesheetPeriodTotalController(ActionRequest request, ActionResponse
 
       BigDecimal periodTotal =
           Beans.get(TimesheetPeriodComputationService.class).computePeriodTotal(timesheet);
+      BigDecimal periodTotalConvert =
+          Beans.get(TimesheetLineService.class).computeHoursDuration(timesheet, periodTotal, false);
+      String periodTotalConvertTitle = timesheetService.getPeriodTotalConvertTitle(timesheet);
 
       response.setAttr("$periodTotalConvert", "hidden", false);
+      response.setAttr("$periodTotalConvert", "value", periodTotalConvert);
+      response.setAttr("$periodTotalConvert", "title", periodTotalConvertTitle);
+
+      Employee employee = timesheet.getEmployee();
+      LocalDate fromDate = timesheet.getFromDate();
+      LocalDate toDate = timesheet.getToDate();
+      String timeUnit = timesheet.getTimeLoggingPreferenceSelect();
+
+      if (employee == null || fromDate == null || toDate == null || StringUtils.isEmpty(timeUnit)) {
+        return;
+      }
+
+      BigDecimal periodTotalLeavesAndHolidays =
+          timesheetService.computeTotalLeavesAndHolidaysForPeriod(
+              employee, fromDate, toDate, timeUnit);
+      BigDecimal periodTotalWorkDurtion =
+          timesheetService.computeTotalWorkDurtionForPeriod(employee, fromDate, toDate, timeUnit);
+
+      response.setAttr("$periodTotalLeavesAndHolidays", "hidden", false);
+      response.setAttr("$periodTotalDueTimeEntries", "hidden", false);
+      response.setAttr("$periodTotalLeavesAndHolidays", "value", periodTotalLeavesAndHolidays);
       response.setAttr(
-          "$periodTotalConvert",
+          "$periodTotalDueTimeEntries",
           "value",
-          Beans.get(TimesheetLineService.class)
-              .computeHoursDuration(timesheet, periodTotal, false));
+          periodTotalWorkDurtion.subtract(periodTotalConvert));
+      response.setAttr(
+          "$periodTotalLeavesAndHolidays",
+          "title",
+          I18n.get("Leaves and public holidays") + " (" + periodTotalConvertTitle + ")");
       response.setAttr(
-          "$periodTotalConvert", "title", timesheetService.getPeriodTotalConvertTitle(timesheet));
+          "$periodTotalDueTimeEntries",
+          "title",
+          I18n.get("Due time entries") + " (" + periodTotalConvertTitle + ")");
```

Comment:
please refactor this method to define a new service to compute these dummy fields.